iree/samples/simple_embedding/simple_embedding_run.cc

```diff
@@ -0,0 +1,221 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A example of setting up the HAL mddule to run simple pointwise array
+// multiplication with the dylib driver.
+#include <stdio.h>
+
+#include "iree/base/api.h"
+#include "iree/hal/api.h"
+
+#if defined(__riscv) && __riscv_xlen == 64
+#include "iree/hal/dylib/registration/driver_module.h"
+#else
+#include "iree/hal/drivers/init.h"
+#endif
+
+#include "iree/modules/hal/hal_module.h"
+#include "iree/vm/api.h"
+#include "iree/vm/bytecode_module.h"
+
+// Compiled module embedded here to avoid file IO:
+#if defined(__riscv) && __riscv_xlen == 64
+#include "iree/samples/simple_embedding/simple_embedding_test_llvm_aot_rv64.h"
+#else
+#include "iree/samples/simple_embedding/simple_embedding_test_bytecode_module.h"
+#endif
```

Comment:
We may need to update https://github.com/google/iree/tree/main/build_tools/embed_data to generate C source instead of C++ to fully switch this sort of sample over to C. I added a note to https://github.com/google/iree/issues/5481.

Reply:
Yes, the PR for c_embed_data is at https://github.com/google/iree/pull/5540. This PR needs to be based on that. A working C sample is at https://github.com/hcindyl/iree/commit/34cca792e3ec0b3de4e2ea896a9a0f81861655e0. I will clean up this branch and PR once https://github.com/google/iree/pull/5540 is merged.